Xamarin.Forms.Platform.Android/Extensions/FragmentManagerExtensions.cs

```diff
@@ -0,0 +1,50 @@
+﻿using FragmentTransaction = Android.Support.V4.App.FragmentTransaction;
+using Fragment = Android.Support.V4.App.Fragment;
+using FragmentManager = Android.Support.V4.App.FragmentManager;
+
+namespace Xamarin.Forms.Platform.Android
+{
+	// This is a way to centralize all fragment modifications which makes it a lot easier to debug
+	internal static class FragmentManagerExtensions
```

Comment:
While this _does_ make the debugging easier, it's also forcing an extra method call on every fragment transaction operation. You could probably achieve the same effect by decorating FragmentTransaction with another class and only using that decorator class when DEBUG is defined. That way you have a central place for debugging but you don't impact the Release version as much.

Something like this:
https://gist.github.com/hartez/a7824a102c595ca096fce8bdca1cec78

Reply:
I think the extra call get optimized out don't they?

![image](https://user-images.githubusercontent.com/5375137/42542693-45a46ea6-8465-11e8-9b3c-c76489572950.png)

I like the idea of the decorator because it also provides a useful place to expand on the debug implementation but I'm concerned about other anomalies of the wrapper or parts of it that can't be overridden like
```c#
public FragmentTransaction SetBreadCrumbShortTitle(string text);
public FragmentTransaction SetBreadCrumbTitle(string text);
```
Though I'm pretty sure these just execute the *ICharSequence* Overrides

Either way I just envision some hard to find inconsistency between debug/release caused by this

Reply:
> Either way I just envision some hard to find inconsistency between debug/release caused by this

That's a reasonable fear. I'm fine with the extension method overhead trade-off for the consistency.